pkg/segment/writer/segstore.go

```diff
@@ -875,26 +871,15 @@ func (segstore *SegStore) computeStarTree() {
 	if segstore.numBlocks == 0 {
 		sortedGrpCols, mCols := segstore.initStarTreeCols()
 		if len(sortedGrpCols) == 0 || len(mCols) == 0 {
-			segstore.usingSegTree = false
 			return
 		}
 
-		hasTreeSpace := false
-		atreeCounterLock.Lock()
-		if currentAgileTreeCount < MaxConcurrentAgileTrees {
-			// for now the first MaxConcurrentAgileTrees segstores will get to AgileTree but
-			// we should add some smart logic on how we can rotate this
-			// amongst other indices/segstores
-			currentAgileTreeCount++
-			hasTreeSpace = true
-		}
-		atreeCounterLock.Unlock()
-		if !hasTreeSpace {
-			segstore.usingSegTree = false
+		segstore.stbHolder = GetSTB()
+		if segstore.stbHolder == nil {
+			log.Infof("computeStarTree: Failed to get STB")
```

Comment:
we should not declare this as a "failed to get" , becayse we may have already used up the 5 concurrent trees, we shuold just treat this as `hasTreeSpace` was false like it was previously

Reply:
I was just logging it for debugging purposes. Addressed in the latest commit.